### ISSUE TITLE
TV3, TV6, Viafree cant play program/episode. Kodi crash

### ISSUE THREAD

**Issue by orfeous1** [retrospect-addon/plugin.video.retrospect#1239]
[kodi.log](https://github.com/retrospect-addon/plugin.video.retrospect/files/3724078/kodi.log)
[retrospect.log](https://github.com/retrospect-addon/plugin.video.retrospect/files/3724079/retrospect.log)

Cant play any episode on TV3 and TV6, Viafree crashes Kodi when trying to open an episode.

Retrospect 5.0.7
Kodi 20191013

**Comment by basrieter** [comment]
So crashing usually I'd not caused by Retrospect but by Kodi internals. As you are running 19-Alpha1 I would suggest you try a stable version of Kodi. 

I see some other Python 3 related issues which need fixing (now that Python 3 is in the nightlies) but not crash related. I will post fixes  for that soon.

**Comment by orfeous1** [comment]
Works great with a week older 19-Alpha1 version though..
I think TV3, TV6.. etc changed something?

**Comment by basrieter** [comment]
Correct, because Python 3 was merged last weekend. That might cause for some issues.

**Comment by basrieter** [comment]
And is it crashing Kodi as an application, or just won't play?

**Comment by basrieter** [comment]
Please use this zip and manually replace the Retrospect files with the newer versions inside the zip. Finally restart Kodi.

[retrospect-patch-20191014.230114.zip](https://github.com/retrospect-addon/plugin.video.retrospect/files/3726399/retrospect-patch-20191014.230114.zip)

**Comment by basrieter** [comment]
Please confirm that this solved your problem @orfeous1

**Comment by orfeous1** [comment]
Patched the file, when trying to start Retrospect it throws an error and exits.
Logfile says something about urihandler and a missing plugin..
attaching logfiles
[retrospect.log](https://github.com/retrospect-addon/plugin.video.retrospect/files/3743567/retrospect.log)

[kodi.log](https://github.com/retrospect-addon/plugin.video.retrospect/files/3743570/kodi.log)

**Comment by basrieter** [comment]
Right! You should install the 5.0.7 from Github and then use this patch file. I forgot to mention that. Sorry.

**Comment by orfeous1** [comment]
Allready running Retrospect 5.0.7

**Comment by orfeous1** [comment]
It seems that when i updated to 5.0.7 via Kodis auto update.. the missing file wasnt included.

**Comment by basrieter** [comment]
I know, but not installed from the zip here in Github. They slightly differ: https://github.com/retrospect-addon/plugin.video.retrospect/releases

**Comment by orfeous1** [comment]
Downloaded 5.0.7 from github, installed, now everything is working with latest Kodi 20191017

**Comment by basrieter** [comment]
Super. I will include it in the next release then.
